operator/cmd/configurator/configurator.go

```diff
@@ -593,35 +605,60 @@ func setAdditionalListeners(additionalListenersCfg, hostIP string, hostIndex int
 		return nil
 	}
 
-	additionalListeners := map[string]string{}
-	err := json.Unmarshal([]byte(additionalListenersCfg), &additionalListeners)
+	additionalListenersCfg, err := utils.Compute(additionalListenersCfg, utils.NewEndpointTemplateData(hostIndex, hostIP, hostIndexOffset), false)
 	if err != nil {
 		return err
 	}
 
-	additionalListenerCfgNames := []string{"redpanda.kafka_api", "redpanda.advertised_kafka_api", "pandaproxy.pandaproxy_api", "pandaproxy.advertised_pandaproxy_api"}
+	structuredDecode := false
```

Comment:
Rather than having a field that could decode in one of two ways, could we instead add a new field that takes precedence over the old one? This behavior will be very confusing to reasonable, especially so if something ever results in an older version of the configuration running with a new version of the operator.